crates/anvil/src/eth/backend/cheats.rs

```diff
@@ -70,4 +93,47 @@ pub struct CheatsState {
     pub impersonated_accounts: AddressHashSet,
     /// If set to true will make the `is_impersonated` function always return true
     pub auto_impersonate_accounts: bool,
+    /// Overrides for ecrecover: Signature => Address
+    pub signature_overrides: HashMap<Bytes, Address>,
+}
+
+/// A custom ecrecover precompile that supports cheat-based signature overrides.
+#[derive(Clone, Debug)]
+pub struct CheatEcrecover {
+    cheats: Arc<CheatsManager>,
+}
+
+impl CheatEcrecover {
+    pub fn new(cheats: Arc<CheatsManager>) -> Self {
+        Self { cheats }
+    }
+}
+
+impl Precompile for CheatEcrecover {
+    fn call(&self, input: PrecompileInput<'_>) -> PrecompileResult {
+        const ECRECOVER_BASE: u64 = 3_000;
```

Comment:
we can also add a check here to return 

ec_recover_run(input.data, input.gas)

early if we dont have any signatures, then this doesnt have any overhead